### PR TITLE
NET-5824: Adding Peers Exported Services API

### PR DESCRIPTION
**Pull request by tauhid621**
### Description

This PR Adds support for listing the exported services to the peers

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: agent/http_register.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ func init() {
 	registerEndpoint("/v1/catalog/node/", []string{"GET"}, (*HTTPHandlers).CatalogNodeServices)
 	registerEndpoint("/v1/catalog/node-services/", []string{"GET"}, (*HTTPHandlers).CatalogNodeServiceList)
 	registerEndpoint("/v1/catalog/gateway-services/", []string{"GET"}, (*HTTPHandlers).CatalogGatewayServices)
+	registerEndpoint("/v1/catalog/exported-services/peers", []string{"GET"}, (*HTTPHandlers).PeerExportedServices)
```
Comment (tauhid621) [inline agent/http_register.go:72]: Not sure if this would be the right API path

File: agent/catalog_endpoint.go [hunk 2]
```diff
@@ -604,3 +604,41 @@ func (s *HTTPHandlers) AssignManualServiceVIPs(resp http.ResponseWriter, req *ht
...
+
+	if err := s.parseEntMeta(req, &args.EnterpriseMeta); err != nil {
+		return nil, err
+	}
+
+	// Make the RPC request
+	var out structs.IndexedExportedServiceList
+	defer setMeta(resp, &out.QueryMeta)
+RPC:
+	if err := s.agent.RPC(req.Context(), "Internal.ExportedPeeredServices", &args, &out); err != nil {
```
Comment (tauhid621) [inline agent/catalog_endpoint.go:626]: Should we use Internal RPCs for public APIs?

Comment (Ganeshrockz) [reply agent/catalog_endpoint.go:626]: Yes. This is how other APIs have been designed

File: agent/http_register.go (hunk 1 above)
Comment (Ganeshrockz) [reply agent/http_register.go:72]: Wondering if we should design the API something like

```
GET /v1/catalog/exported-services?partition=<part>
GET /v1/catalog/exported-services?peer=<peer-name>
```

Both `part` and `peer-name` can also support wildcards.

The response could look something like

```json
[
  {
      "Partition": "Part-1",
      "Services": [....]
  },
  {
      "Peer": "Peer-1",
      "Services": [....]
  }
]
```

depending on the inputs. What do you think?

**Comment by tauhid621** [comment]
Closing this PR as the exported services are already part of the peering APIs

**Pull request closed without merging**